rfcs/0169-feature-parameter-names.md

```diff
@@ -0,0 +1,537 @@
+---
+feature: feature_parameter_names
+start-date: 2024-01-10
+author: Dmitry Bogatov
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Nixpkgs has mechanisms to exports compile-time options of the upstream build
+system through the named parameters of derivation function, but the names of
+these parameters are inconsistent from package to package and from feature to
+feature.
+
+# Motivation
+[motivation]: #motivation
+
+Developer interface for building environment with specific features enabled and
+disabled is more complicated and requires more knowledge of implementation
+details compared to already existing systems.
+
+For example, Gentoo Linux has exhaustive list of [all known configuration
+flags](https://www.gentoo.org/support/use-flags) and has means to
+programmatically query what flags are available for particular package. Nixpkgs
+has neither, and this RFC strives to rectify it.
```

Comment:
You seem to describe the problem of `.override` attributes not being easily discoverable, but the RFC doesn't actually address that at all from what I understand.

If that's the actual problem you're trying to solve, the easiest solution might be to make those attributes more easily discoverable, e.g. by rendering attributes with their documentation on search.nixos.org. Or on the deep end, something like https://discourse.nixos.org/t/working-group-member-search-module-system-for-packages/26574

However, the actual problem you seem to want to solve with this RFC is described better in your comment here: https://github.com/NixOS/nixpkgs/pull/234463#issuecomment-1574892207. If that's indeed the case, the motivation section should be rewritten to talk about that.

Reply:
It actually does address, since now feature parameters are clearly namespaced from all other. To make it clearer,
I rewrote the motivation part as list of good things that we will get by implementing this RFC. Thoughts?